decode/loadjpeg.c

```diff
@@ -74,7 +75,9 @@ int convert_one_image(const char *infilename)
   buf = (unsigned char *)malloc(length_of_file + 4);
   if (buf == NULL)
     exitmessage("Not enough memory for loading file\n");
-  fread(buf, length_of_file, 1, fp);
+  n_items = fread(buf, length_of_file, 1, fp);
+  if (n_items != 1)
+      exitmessage("Reading file fail\n");
```

Comment:
You should also `free(buf);` and `fclose(fp);` if the fread fails

Reply:
Will close the fd, tks